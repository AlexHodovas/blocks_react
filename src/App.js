import React from "react";
import Tables from "../src/components/tables/Tables";

export const amountTables = +process.env.REACT_APP_AMOUNT_TABLES;
export const initialWidth = +process.env.REACT_APP_INITIAL_WIDTH;
export const initialHeight = +process.env.REACT_APP_INITIAL_HEIGHT;
export const cellSize = +process.env.REACT_APP_CELL_SIZE;

const App = () => <Tables />;

export default App;
