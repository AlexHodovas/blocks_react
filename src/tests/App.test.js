import React from "react";
import { configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import App, { initialWidth, initialHeight, cellSize } from "../App";

configure({ adapter: new Adapter() });

it("full App renders correctly", () => {
  const tree = renderer
    .create(
      <App
        initialWidth={initialWidth}
        initialHeight={initialHeight}
        cellSize={cellSize}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe("buttons", () => {
  const onClick = jest.fn();
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<App onClick={onClick} />);
  });

  it("addRowButton works", () => {
    const button = wrapper.find("#addRowButton").at(0);
    button.simulate("click");

    expect(
      wrapper
        .find("#addRowButton")
        .at(0)
        .props().tableHeight
    ).toEqual(5);
  });

  it("addColumnButton works", () => {
    const button = wrapper.find("#addColumnButton").at(0);
    button.simulate("click");

    expect(
      wrapper
        .find("#addColumnButton")
        .at(0)
        .props().tableWidth
    ).toEqual(5);
  });

  it("deleteRowButton works", () => {
    const button = wrapper.find("#deleteRowButton").at(0);
    button.simulate("click");

    expect(
      wrapper
        .find("#deleteRowButton")
        .at(0)
        .props().tableHeight
    ).toEqual(3);
  });

  it("deleteColumnButton works", () => {
    const button = wrapper.find("#deleteColumnButton").at(0);
    button.simulate("click");

    expect(
      wrapper
        .find("#deleteColumnButton")
        .at(0)
        .props().tableWidth
    ).toEqual(3);
  });

  it("last row not removed", () => {
    const button = wrapper.find("#deleteRowButton").at(0);
    button.simulate("click");
    button.simulate("click");
    button.simulate("click");

    expect(
      wrapper
        .find("#deleteRowButton")
        .at(0)
        .props()
        .isOnlyOneRow()
    ).toEqual(true);

    expect(
      wrapper
        .find("#deleteRowButton")
        .at(0)
        .props().isDeleteRowButtonHidden
    ).toEqual(true);
  });

  it("last column not removed", () => {
    const button = wrapper.find("#deleteColumnButton").at(0);
    button.simulate("click");
    button.simulate("click");
    button.simulate("click");

    expect(
      wrapper
        .find("#deleteColumnButton")
        .at(0)
        .props()
        .isOnlyOneColumn()
    ).toEqual(true);

    expect(
      wrapper
        .find("#deleteColumnButton")
        .at(0)
        .props().isDeleteColumnButtonHidden
    ).toEqual(true);
  });
});
