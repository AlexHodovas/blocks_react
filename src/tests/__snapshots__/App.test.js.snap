// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`full App renders correctly 1`] = `
<div
  className="sc-AxiKw eoHjAy"
  onDragStart={[Function]}
  onMouseDown={[Function]}
>
  <table
    className="table"
    onMouseEnter={[Function]}
  >
    <tbody>
      <tr
        onMouseOver={[Function]}
      >
        <td
          className="td"
          onMouseOver={[Function]}
        />
        <td
          className="td"
          onMouseOver={[Function]}
        />
        <td
          className="td"
          onMouseOver={[Function]}
        />
        <td
          className="td"
          onMouseOver={[Function]}
        />
      </tr>
      <tr
        onMouseOver={[Function]}
      >
        <td
          className="td"
          onMouseOver={[Function]}
        />
        <td
          className="td"
          onMouseOver={[Function]}
        />
        <td
          className="td"
          onMouseOver={[Function]}
        />
        <td
          className="td"
          onMouseOver={[Function]}
        />
      </tr>
      <tr
        onMouseOver={[Function]}
      >
        <td
          className="td"
          onMouseOver={[Function]}
        />
        <td
          className="td"
          onMouseOver={[Function]}
        />
        <td
          className="td"
          onMouseOver={[Function]}
        />
        <td
          className="td"
          onMouseOver={[Function]}
        />
      </tr>
      <tr
        onMouseOver={[Function]}
      >
        <td
          className="td"
          onMouseOver={[Function]}
        />
        <td
          className="td"
          onMouseOver={[Function]}
        />
        <td
          className="td"
          onMouseOver={[Function]}
        />
        <td
          className="td"
          onMouseOver={[Function]}
        />
      </tr>
    </tbody>
  </table>
  <div
    className="sc-AxjAm ebUBhh"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      +
    </button>
  </div>
  <div
    className="sc-AxirZ jprgzN"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      +
    </button>
  </div>
  <div
    className="sc-AxhUy kqetBS"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      -
    </button>
  </div>
  <div
    className="sc-AxhCb dgwgFs"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      -
    </button>
  </div>
</div>
`;
